### PR TITLE
Add gvamono3d: monocular 3D object detection element

### PR DESCRIPTION
**Pull request by msmiatac**
**Summary**

Adds a new inference element, gvamono3d, that performs monocular 3D object detection (e.g. MonoDETR) from a single camera image plus its calibration, and emits typed camera-frame 3D metadata (GstAnalyticsCamera3DODMtd). The existing gvawatermark3d renderer is reworked into a VA-aware bin so it can consume that typed metadata and draw 3D cuboids, and the JSON metaconvert path is extended to serialize the 3D fields. A model-conversion script for MonoDETR is included.

**Motivation**

DL Streamer had no path for camera-based 3D object detection. Rather than a standalone GstBaseTransform, gvamono3d is implemented as a thin GvaBaseInference subclass (like gvadetect), so it reuses the shared inference engine: batching, VA pre-processing, model-instance sharing, multi-stream, and the standard inference properties.

**What's included**

New element gvamono3d (gvamono3d)
- Subclass of GvaBaseInference (GST_GVA_MONO3D_TYPE); properties calibration-file and threshold on top of the inherited inference properties.
- Driven entirely from the model's rt_info (model_type=mono3d) — no model-proc file required.
- Registered as gvamono3d in the videoanalytics plugin.

3D post-processing (common/post_processor/converters/to_roi/mono3d.)

- Mono3DConverter decodes MonoDETR's 5 outputs (logits, 2D box cxcylrtb, 3D dims, depth+sigma, heading bins/residuals), back-projects with the calibration P2, and produces a 2D box + 3D fields per object. Handles FP16/FP32 outputs.
- New metadata GstAnalyticsCamera3DODMtd (include/dlstreamer/gst/metadata/camera_3d_od_mtd) carrying location, dimensions, orientation and alpha, linked to the 2D od_mtd.

Calibration (common/mono3d_calibration)
- calibration-file accepts KITTI .txt (P2: row) or JSON (intrinsic_matrix 3x3 / projection_matrix 3x4, optional image_size).

Multi-input model plumbing (inference_impl.cpp, pre_processors.cpp, openvino_image_inference.*, image_inference.)

- Registers the auxiliary calib and img_sizes inputs (in addition to image) and fills them at pre-process time.
- Pins dynamic input batch dims after ov::set_batch to avoid Concat shape crashes on MonoDETR.
- Auto-applies EXECUTION_MODE_HINT=ACCURACY on GPU (default performance mode overflows the FP16 backbone → NaNs); overridable via ie-config.

Rendering (gvawatermark3d)

- Reworked into a VA-aware GstBin ([vapostproc] → videoconvert → BGR → render → videoconvert) with a hidden inner filter GstGvaWatermark3DRender, so it accepts VA/DMABuf/system input from decodebin3 and draws 3D boxes from the typed metadata. Forwards intrinsics-file/calibration-file.

Serialization (gvametaconvert/jsonconverter.cpp)

- Emits the 3D fields (location, dimensions, orientation) to JSON.

Model conversion (convert_monodetr.py + requirements_convert_monodetr.txt)
- Converts a MonoDETR checkpoint to OpenVINO IR with the required rt_info.

Docs

- New page gvamono3d.md and an index row in elements.md.

**Example**

```
gst-launch-1.0 filesrc location=input.mp4 ! decodebin3 ! \
  gvamono3d model=monodetr.xml device=GPU calibration-file=000000.txt threshold=0.2 ! \
  gvawatermark3d calibration-file=000000.txt ! videoconvert ! autovideosink
```

### CODE REVIEW

**Comment by msmiatac** [comment]
@copilot resolve the merge conflicts in this pull request
